superset/security/manager.py

```diff
@@ -2473,32 +2481,73 @@ def get_rls_filters(self, table: "BaseDatasource") -> list[SqlaQuery]:
             return []
 
         # pylint: disable=import-outside-toplevel
+        from flask_appbuilder.security.sqla.models import assoc_group_role
+
         from superset.connectors.sqla.models import (
+            RLSFilterGroups,
             RLSFilterRoles,
             RLSFilterTables,
             RowLevelSecurityFilter,
         )
 
+        # Roles and Groups directly linked to the user
         user_roles = [role.id for role in self.get_user_roles(g.user)]
```

Comment:
`get_user_roles(g.user)` will get all roles that are assigned to a user, directly or indirectly via groups